ccm_web/server/src/canvas/canvas.service.ts

```diff
@@ -14,11 +14,51 @@ import { User } from '../user/user.model'
 import { CanvasConfig } from '../config'
 import { DatabaseError } from '../errors'
 import baseLogger from '../logger'
+import { IncomingHttpHeaders } from 'http'
 
 const logger = baseLogger.child({ filePath: __filename })
 
 type SupportedAPIEndpoint = '/api/v1/' | '/api/graphql/'
-const requestorOptions: GotOptions = { retry: { limit: 2, methods: ['POST', 'GET', 'PUT', 'DELETE'] } }
+
+interface IncomingRateLimitedCanvasHttpHeaders extends IncomingHttpHeaders {
+  'x-rate-limit-remaining'?: string
+  'x-request-cost'?: string
+}
+
+const requestorOptions: GotOptions = {
+  retry: {
+    limit: 2,
+    methods: ['POST', 'GET', 'PUT', 'DELETE'],
+    // TODO: After got@12 upgrade, replace following list with something like…
+    // got.defaultInternals.retry.statusCodes.concat(403)
+    statusCodes: [403, 408, 413, 429, 500, 502, 503, 504, 521, 522, 524]
```

Comment:
There is a `calculateDelay` option that allows for some pretty sophisticated logic on spacing out the retries. I did one test run with 400 enrollments with the options as is, and 13 of them failed. Then, I tried adding a very simplistic value for `calculateDelay` of `() => { return 1000 }` and all 400 of them went through. Not even sure a full second is necessary. Might be worth adding something like this while we discuss throttling, etc.